src/encoded/visualization.py

```diff
@@ -2440,7 +2460,7 @@ def generate_batch_hubs(context, request):
             if g_text == '':
                 log.debug('Requesting %s.%s#%s NO ASSEMBLY !!!' % (page,suffix,cmd))
                 g_text = json.dumps(results,indent=4)
-                assemblies = [result['assemblies'] for result in results['@graph']]
+                assemblies = [result.get('assemblies',[]) for result in results['@graph']]
```

Comment:
Not sure that this is a more straightforward option.  I think this best change is actually to rip this code out as it only feeds a log.debug message that is never even seen.

Reply:
Comment on your example rewrite below.  This would need further debugging, I can see at least one error, but that is understandable as I can't expect that you had a chance to run it.  While I can see the value of attempting to eliminate duplicate code and moving certain conditional statements outside loops so they are more efficient, I do not agree that your version represents an improvement for a couple of reasons.

1) You replace logic inefficiency with storage inefficiency.  Declaring variables to hold conditional results is not free.  In the case of quickview_blocked, for example, there are currently no more than 2 assemblies per experiment, so you replaced 2 conditional statements with 2 conditional statements.
2) You obscure the meaning of code when trying to make it more efficient.  While far from crystal clear, I think there was more simplicity in the original code due to the separation of the three types: ucsc, ensembl and quickview.  This code will certainly have to be touched again, for instance if a 4th option were supported, and your version will make that a little harder.
3) Your method requires going though all files, which might be a lengthy list, where as the original method could shorten that search.  Of course the original method might actually go through the files list twice, stopping on the first hit each time. 

Anyway, it would have been nice to do this face to face.  Do not misunderstand me, I do appreciate ALL your suggestions, even if some of them might generate debate.